### PR TITLE
feat(Scanner): Add Scanner module

### PR DESCRIPTION
**Pull request by Crash--**
Create a new "Scanner" module. 

Used in Drive ATM 

- [x] Use withOffline from UI (need to be done) 
- [x] Use ExperimentalModal instead of Modal (semantic release didn't create a release with the new  style...) 
- [x] Write a few tests at least for getter

### CODE REVIEW

File: packages/cozy-scanner/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "cozy-scanner",
+  "version": "0.0.1",
+  "description": "Cozy-Scanner is used to access device comera and also provide component to categorize document",
```
Comment (ptbrowne) [inline packages/cozy-scanner/package.json]:
```suggestion
  "description": "Cozy-Scanner provides a component to take a picture of a document and categorize it",
```

File: packages/cozy-scanner/src/DocumentCategory.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+/**
+ * DocumentCategory component
+ *
+ * The goal of this component is to display a
+ * category / type of document and also manage
+ * its associated ActionMenu since a category has
+ * several associated items.
+ *
+ * If an item from a category is selected, we display
+ * its label instead of the label from the category
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentCategory.jsx]: Is it still true ? Don't we show two lines now ?

File: packages/cozy-scanner/src/DocumentCategory.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+} from 'cozy-ui/transpiled/react/ActionMenu'
+import palette from 'cozy-ui/transpiled/react/palette'
+import GridItem from 'cozy-ui/transpiled/react/Labs/GridItem'
+
+import CategoryGridItem from './CategoryGridItem'
+
+import styles from './styles.styl'
+
+/**
+ * DocumentCategory component
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentCategory.jsx]: I think this line is not necessary.

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+ * Document Qualification
+ *
+ * This screen is used to qualify a document ie:
+ *  Selecting a metadata category etc
+ *  Renaming the file
+ *
+ * When a selection is done, we call a callback from
+ * its parent
+ *
+ */
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]: The docstring is not attached to the component.

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            <InputGroup
+              fullwidth
+              append={<span className="u-pr-1">{filename_extension}</span>}
+              className="u-bdrs-3"
+            >
+              <Input
+                placeholder={t('Scan.filename')}
+                value={filename}
+                onChange={event => {
+                  //If the user write something once, we don't want to rename the file automatically anymore
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
                  // If the user writes something once, we don't want to rename the file automatically anymore
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                  if (!hasUserWrittenFileName) {
+                    this.setState({ hasUserWrittenFileName: true })
+                  }
+                  //If we left an empty value, then we reset the behavior
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
                  // If we left an empty value, then we reset the behavior
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+DocumentQualification.defaultProps = {
+  editFileName: false
+}
+DocumentQualification.propTypes = {
+  /**
+   * This callback is called after a select.
+   *
+   */
+  onQualified: PropTypes.func,
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
  /** Called after category selection */
  onQualified: PropTypes.func,
```

Then why isn't the name onCategorySection ;)

File: packages/cozy-scanner/src/DocumentTypeData.js [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+    icon: 'bank',
+    file_type_ids: ['34', '35', '36']
+  },
+  {
+    id: 'theme8',
+    label: 'invoice',
+    icon: 'euro',
+    file_type_ids: ['27', '28', '29', '26', '30']
+  }
+]
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentTypeData.js]: Nit: Could be a JSON. Helpful when working with jq.  Not mixing data with code :+1:

File: packages/cozy-scanner/src/DocumentTypeData.js [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+        items.push(getItemById(itemId))
+      })
+    }
+  })
+  return items
+}
+
+export const getItemById = id => {
+  return items.find(item => item.id === id)
+}
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentTypeData.js]: When calliing a method like that, I would not expect it to be O(n). Myabe you can build a indexed view of items ?

```
const itemsById = keyBy(items, item => item.id)
```

File: packages/cozy-scanner/src/DocumentTypeData.js [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+export const getThemeByItem = item => {
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentTypeData.js]:
```suggestion
export const getItemTheme = item => {
```

File: packages/cozy-scanner/src/DocumentTypeData.js [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+  if (item.defaultTheme) {
+    return themes.find(theme => theme.id === item.defaultTheme)
+  }
+  return themes.find(theme => theme.file_type_ids.includes(item.id))
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentTypeData.js]: We have to scan in O(n*m) here to find the theme of an item.

I think it would be better (more performant, and the data would be clearer) if the data was reorganized. The theme could be flattened in the items at launch time for example.

File: packages/cozy-scanner/src/Scanner.jsx [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+    const { generateName } = this.props
+    const name = filename === '' ? generateName() : filename
+    this.setState({ status: SCANNER_UPLOADING, filename: name })
+    const { dirId, onConflict, onBeforeUpload, onFinish } = this.props
+    const onResolvedLocalFS = async fileEntry => {
+      fileEntry.file(
+        file => {
+          const reader = new FileReader()
+          reader.onloadend = async () => {
+            //we get the of the readAsBuffer in the `result` attr
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: What ? ;)

File: packages/cozy-scanner/src/Scanner.jsx [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+          onConflict,
+          metadata
+        )
+      }
+    } catch (error) {
+      if (/Not Found/.test(error)) {
+        return await this.upload(name, file, dirId, metadata)
+      }
+      throw error
+    }
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: IMHO the two methods, before being extracted to FileCollection, should be extracted from the component.

File: packages/cozy-scanner/src/Scanner.jsx [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+      metadata
+    })
+  }
+  onFail = message => {
+    this.setState({ loadingScreen: false })
+    console.log('failed', message)
+  }
+  startScanner = () => {
+    try {
+      //
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: Extra comment

File: packages/cozy-scanner/src/Scanner.jsx [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+    )
+  }
+}
+
+Scanner.defaultProps = {
+  pluginConfig: {}
+}
+/**
+ *
+ */
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: Extra comment.

File: packages/cozy-scanner/src/ScannerQualification.jsx [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+import { translate } from 'cozy-ui/transpiled/react/I18n'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+
+import DocumentQualification from './DocumentQualification'
+
+import { Modal as NewModal } from 'cozy-ui/transpiled/react'
+
+import { getTracker } from 'cozy-ui/transpiled/react/helpers/tracker'
+
+const pushAnalytics = qualification => {
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/ScannerQualification.jsx]:
```suggestion
const trackQualification = qualification => {
```

File: packages/cozy-scanner/src/ScannerQualification.jsx [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+        'Qualification',
+        qualification.label
+      ])
+    }
+  }
+}
+/**
+ * ScannerQualification component
+ *
+ * This component is used to be displayed just after taking a
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/ScannerQualification.jsx]:
```suggestion
 * Displayed just after taking a
```

File: packages/cozy-scanner/src/ScannerQualification.jsx [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+      />
+    )
+  }
+}
+
+ScannerQualification.propTypes = {
+  /**
+   * primary action of the modal
+   *
+   */
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/ScannerQualification.jsx]: Can you use comments like this `/** Primary action of the modal */` ?

It is a bit weird here since we do not know that there is a modal at this point.

File: packages/cozy-scanner/src/locales/en.json [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+  "Scan": {
+    "scan_a_doc": "Scan a doc",
+    "save_doc": "Save the doc",
+    "filename": "Filename",
+    "save": "Save",
+    "cancel": "Cancel",
+    "qualify": "Categorize",
+    "apply": "Apply",
+    "error": {
+      "offline": "You are currently offline and you can't use this functionnality. Try it later",
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/locales/en.json]:
```suggestion
      "offline": "You are currently offline and you can't use this functionality. Please, try again later",
```

File: packages/cozy-scanner/src/locales/en.json [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+      "uploading": "You are already uploading a file. Wait until the end of this upload and try again.",
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/locales/en.json]:
```suggestion
      "uploading": "You are already uploading a file. Please wait until the end of this upload and try again.",
```

File: packages/cozy-scanner/src/locales/en.json [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+      "generic": "Something went wrong. Please try again."
+    },
+    "successful": {
+      "qualified_ok": "You just have successfully categorized your file! "
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/locales/en.json]: Sometimes we see qualified, sometimes "categorized", it seems confusing.

File: packages/cozy-scanner/src/DocumentQualification.jsx (hunk 6 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentQualification.jsx]: we should have this rule in our eslint / prettier config if you want one space after `//`.

File: packages/cozy-scanner/src/DocumentQualification.jsx (hunk 7 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentQualification.jsx]: Because now we can also changed the filename. Updated the comment

Comment (Crash--) [reply packages/cozy-scanner/src/DocumentQualification.jsx]: Hmm, in fact, since I also have a callback for the onFileNameChange, I think I can remove the stuff related to filename and rename it to onCategorySelection 🤔

File: packages/cozy-scanner/src/DocumentTypeData.js (hunk 11 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentTypeData.js]: Modern JS engine has optimization for this kind of behavior even if I agree that we can flattened the datas. Will see what I can do

File: packages/cozy-scanner/src/Scanner.jsx (hunk 13 above)
Comment (Crash--) [reply packages/cozy-scanner/src/Scanner.jsx]: In fact there are already extracted. I need to use them ;)

File: packages/cozy-scanner/src/ScannerQualification.jsx (hunk 18 above)
Comment (Crash--) [reply packages/cozy-scanner/src/ScannerQualification.jsx]: I think I don't get your point. 

We are in a component calling `ExperimentalModal`. We know that there is a modal at this point, no?

File: packages/cozy-scanner/src/locales/en.json (hunk 21 above)
Comment (Crash--) [reply packages/cozy-scanner/src/locales/en.json]: We now use `describe` in English

File: packages/cozy-scanner/src/DocumentQualification.spec.jsx [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+  global.Date.parse = _Date.parse
+  global.Date.now = _Date.now
+}
+
+const unMockDate = () => {
+  global.Date = new Date()
+  global.Date.UTC = Date.UTC
+  global.Date.parse = Date.parse
+  global.Date.now = Date.now
+}
```
Comment (y-lohse) [inline packages/cozy-scanner/src/DocumentQualification.spec.jsx]: Why not use [`mockdate`](https://github.com/konnectors/toutatice/blob/d983393ff8132579a6849f8996a2a8d2cf7a8982/src/helpers/transpileGroupToCozy.spec.js#L8-L14) instead?

File: packages/cozy-scanner/src/ScannerQualification.jsx (hunk 18 above)
Comment (y-lohse) [reply packages/cozy-scanner/src/ScannerQualification.jsx]: Technically we're calling `ScannerQualification` when using these proptypes :)

File: packages/cozy-scanner/Readme.md [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+Cozy-Scanner requires :
+
+- a cordova plugin : `cordova plugin add https://github.com/cozy/cordova-plugin-camera.git#bf519ca009eeaec3a47237ee07a09d515cc733df`
+- At least Cozy-UI 25.7
+- Cozy-Doctypes
+
+To use the component and get the style, you need to import the css in your app
+
+```js
+import "cozy-scanner/dist/stylesheet.css";
```
Comment (ptbrowne) [inline packages/cozy-scanner/Readme.md]: It would be nice not to have to do that ? Why is this necessary ? Could the scanner component add this stylesheet by itself ?

File: packages/cozy-scanner/Readme.md [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+Since Cozy-Scanner uses a few `Aleter` you also have to put `<Alerter />` (from Cozy-UI)
```
Comment (ptbrowne) [inline packages/cozy-scanner/Readme.md]:
```suggestion
Since Cozy-Scanner uses a few `Alerter` you also have to put `<Alerter />` (from Cozy-UI)
```

File: packages/cozy-scanner/babel-transformer.js [hunk 25]
```diff
@@ -0,0 +1,3 @@
+const config = require('./babel.config')
+const { createTransformer } = require('babel-jest')
+module.exports = createTransformer(config)
```
Comment (ptbrowne) [inline packages/cozy-scanner/babel-transformer.js]: Why is this necessary ? Can you comment ? Is this because we use a custom babel config ?

File: packages/cozy-scanner/src/CategoryGridItem.jsx [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <div
+      className={classNames(
+        styles['grid-item'],
+        'u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer',
+        {
+          [styles['border-selected']]: isSelected,
+          [styles['grid-item__selected']]: isSelected && label,
+          [styles['grid-item__selected__without_label']]: isSelected && !label,
+          [styles['border-not-selected']]: !isSelected
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/CategoryGridItem.jsx]: It seems a `GridItem` that would care about the `isSelected` prop could be extracted here.

It seems abnormal to have to put 4 different classes here also :/

File: packages/cozy-scanner/src/DocumentCategory.jsx [hunk 27]
```diff
@@ -0,0 +1,124 @@
...
+    const { onSelect } = this.props
+    if (onSelect) onSelect(item)
+  }
+  render() {
+    const { isMenuDisplayed } = this.state
+    const { category, isSelected, selectedItem, items, t } = this.props
+    return (
+      <>
+        <GridItem onClick={() => this.toggleMenu()}>
+          <CategoryGridItem
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentCategory.jsx]: CategoryBox maybe ?

It was unclear to me at this point that GridItem was only dealing with the Layout where CategoryGridItem deals with the border and the selection.

File: packages/cozy-scanner/src/DocumentCategory.spec.jsx [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+          t={text => text}
+        />
+      </MuiCozyTheme>
+    )
+    expect(asFragment()).toMatchSnapshot()
+    //Click on the item
+    fireEvent.click(getByText('Scan.items.test'))
+    expect(asFragment()).toMatchSnapshot()
+    //Check if the second item is displayed in the ActionMenu
+    expect(getByText('Scan.items.Label2'))
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentCategory.spec.jsx]: Seems like you do not expect anything here.

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 29]
```diff
@@ -0,0 +1,206 @@
...
+import { getItemsByCategory } from './DocumentTypeDataHelpers'
+import GridItem from 'cozy-ui/transpiled/react/Labs/GridItem'
+import styles from './styles.styl'
+
+const filename_extension = '.jpg'
+const id_filename_input = 'filename_input'
+
+/**
+ *
+ * This screen is used to qualify a document ie:
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
 * Used to qualify a document ie:
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 30]
```diff
@@ -0,0 +1,206 @@
...
+ *  Selecting a metadata category etc
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
 *   - Selecting a metadata category etc
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 31]
```diff
@@ -0,0 +1,206 @@
...
+ *  Renaming the file
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
 *   - Renaming the file
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 32]
```diff
@@ -0,0 +1,206 @@
...
+ *
+ * When a selection is done, we call a callback from
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]:
```suggestion
 * When a selection is done, `props.onDescribed` is called with the selected item
```

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 33]
```diff
@@ -0,0 +1,206 @@
...
+   *
+   */
+  onDescribed: PropTypes.func,
+  onFileNameChanged: PropTypes.func,
+  title: PropTypes.string,
+  initialSelected: PropTypes.shape({
+    itemId: PropTypes.string,
+    categoryLabel: PropTypes.string
+  }),
+  editFileName: PropTypes.bool
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]: It is unclear to me from the name what this prop does.

File: packages/cozy-scanner/src/DocumentQualification.jsx [hunk 34]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+DocumentQualification.defaultProps = {
+  editFileName: false
+}
+DocumentQualification.propTypes = {
+  /**
+   * This callback is called after a qualification
+   *
+   */
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.jsx]: Can you use "shorthand" docstrings ? `/** Called after qualification (with what ?) */`

File: packages/cozy-scanner/src/DocumentQualification.spec.jsx [hunk 35]
```diff
@@ -0,0 +1,85 @@
...
+describe('DocumentQualification', () => {
+  it('Inital render + selection + filename', () => {
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/DocumentQualification.spec.jsx]:
```suggestion
  it('Initial render + selection + filename', () => {
```

File: packages/cozy-scanner/src/Scanner.jsx [hunk 36]
```diff
@@ -0,0 +1,190 @@
...
+import { CozyFile } from 'cozy-doctypes'
+import { Modal } from 'cozy-ui/transpiled/react'
+
+import ScannerQualification from './ScannerQualification'
+import withOffline from 'cozy-ui/transpiled/helpers/withOffline'
+export const SCANNER_IDLE = 'idle'
+export const SCANNER_DONE = 'done'
+export const SCANNER_UPLOADING = 'uploading'
+/**
+ * LoadingScreen is used to create a better transition between
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: LoadingScreen ?

File: packages/cozy-scanner/src/Scanner.jsx [hunk 37]
```diff
@@ -0,0 +1,190 @@
...
+ * the native camera and the Qualification Modale.
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]:
```suggestion
 * the native camera and the qualification modal.
```

File: packages/cozy-scanner/src/Scanner.jsx [hunk 38]
```diff
@@ -0,0 +1,190 @@
...
+          // Read the file as an ArrayBuffer
+          reader.readAsArrayBuffer(file)
+        },
+        err => {
+          this.setState({ error: err })
+          this.setState({ status: SCANNER_DONE })
+          //Since this module is pretty recent, let's have this info in sentry if needed
+          console.error('error getting fileentry file!' + err) //eslint-disable-line no-console
+        }
+      )
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/Scanner.jsx]: This function seems pretty complex and is not tested as far as I am aware. It seems to deal with several things.

- state of the component
- reading the file
- uploading the file

Do you think we could we extract the file reading and uploading logic out the component so that the component only deal with its state and props ?

```
     this.setState({ status: SCANNER_UPLOADING, filename: name })
     try {
     ... do before upload stuff ...
      const { data: newFile } = await doUpload(imageURI)
      this.setState({ filename: newFile.name }) 
     } catch (e) {
       this.setState({ status: SCANNER_DONE, error: e }) 
     } finally {
       this.setState({ status: SCANNER_DONE }) 
      }
```

File: packages/cozy-scanner/src/ScannerQualification.jsx [hunk 39]
```diff
@@ -0,0 +1,90 @@
...
+        dismissAction={dismissAction}
+        primaryText={t('Scan.save')}
+        primaryAction={async () => {
+          pushAnalytics(qualification)
+          try {
+            await onSave(qualification, filename)
+          } catch (error) {
+            Alerter.error(t('Scan.error.generic'))
+          }
+        }}
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/ScannerQualification.jsx]: You might want to put the handler creation outside the render so that children components can bypass render with PureComponent.

File: packages/cozy-scanner/src/ScannerQualification.jsx (hunk 18 above)
Comment (ptbrowne) [reply packages/cozy-scanner/src/ScannerQualification.jsx]: Yes for the user/reader, when reading the prop types, (s)he might not know that the implementation is through a modal. If the ScannerQualification component had `Modal` in its name, I think it would be clearer.

File: packages/cozy-scanner/src/DocumentQualification.spec.jsx (hunk 22 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentQualification.spec.jsx]: Thanks

File: packages/cozy-scanner/Readme.md (hunk 23 above)
Comment (Crash--) [reply packages/cozy-scanner/Readme.md]: At the beginning I wanted to let the app dealing with the style without having to import a css file. But `scanner` use a few stylus files, and this means the app needs to be able to handle stylus file and this is something I want to avoid for the sake of simplicity to use. 

So the option was to generate a stylesheet.css file. At this moment, we have two choice : 
1/ Having each of our component importing this stylesheet 
2/ Let the app importing the stylesheet 

Again, I choose the 2 because copy / pasting a line, is simpler for DX. Having the same stylesheet imported in each component will result in a `big bundle size` (right?) or... you'll need to customize your build tools and this is something I try to avoid.

File: packages/cozy-scanner/babel-transformer.js (hunk 25 above)
Comment (Crash--) [reply packages/cozy-scanner/babel-transformer.js]: Comment added: 

/**
 * Since we are using a custom babel config
 * we need to create a Transformer to tell
 * jest to use it
 */

File: packages/cozy-scanner/src/DocumentCategory.spec.jsx (hunk 28 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentCategory.spec.jsx]: No. I expect to see the ActionMenu. And to to that, I check if I see this label

File: packages/cozy-scanner/src/DocumentQualification.jsx (hunk 33 above)
Comment (Crash--) [reply packages/cozy-scanner/src/DocumentQualification.jsx]: renamed to allowEditFilename

File: packages/cozy-scanner/Readme.md (hunk 23 above)
Comment (ptbrowne) [reply packages/cozy-scanner/Readme.md]: > Having the same stylesheet imported in each component will result in a big bundle size (right?) 

No. The stylesheet will be only once in the bundle.

File: packages/cozy-scanner/src/styles.styl [hunk 40]
```diff
@@ -0,0 +1,35 @@
...
+    padding-top rem(12)
+    }
+
+.grid-item-theme {
+    font-size rem(12)
+    padding-top rem(4)
+    }
+.grid-item-label {
+    font-size rem(10)
+    }
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/styles.styl]: Closing brackets are indented, could you fix it please ;)

File: packages/cozy-scanner/Readme.md (hunk 23 above)
Comment (Crash--) [reply packages/cozy-scanner/Readme.md]: Removed the need of stylus since the CSS is pretty easy and having to import manually the stylesheet in the transpiled code was not very easy

File: packages/cozy-scanner/src/CategoryGridItem.jsx [hunk 41]
```diff
@@ -4,7 +4,8 @@ import classNames from 'classnames'
 import palette from 'cozy-ui/transpiled/react/palette'
 import { Icon, IconStack } from 'cozy-ui/transpiled/react'
 
-import styles from './styles.styl'
+import styles from './stylesheet.css'
+console.log('styles', styles)
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/CategoryGridItem.jsx]: console.log to remove

File: packages/cozy-scanner/src/stylesheet.css [hunk 42]
```diff
@@ -0,0 +1,30 @@
...
+.grid-item__selected {
+  padding-top: 0.75rem;
+}
+.grid-item-theme {
+  font-size: 0.75rem;
+  padding-top: 0.25rem;
+}
+.grid-item-label {
+  font-size: 0.625rem;
+}
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/stylesheet.css]: This does not look specific to the scanner ? Can you add a comment so that we do not forget that this should be in UI ?

File: packages/cozy-scanner/src/CategoryGridItem.jsx (hunk 41 above)
Comment (Crash--) [reply packages/cozy-scanner/src/CategoryGridItem.jsx]: already done

File: packages/cozy-scanner/src/stylesheet.css [hunk 43]
```diff
@@ -0,0 +1,30 @@
...
+  is not well centered. So I can't just rem(16) ... */
+.grid-item-title {
+  margin-top: 1.188rem;
+}
+.grid-item__selected__without_label {
+  padding-top: 1rem;
+}
+.grid-item__selected {
+  padding-top: 0.75rem;
+}
```
Comment (ptbrowne) [inline packages/cozy-scanner/src/stylesheet.css]: Why a selected grid-item has a different padding-top ?

Comment (Crash--) [reply packages/cozy-scanner/src/stylesheet.css]: because we dont display it in the same way...
